nucypher/cli/commands/deploy.py

```diff
@@ -313,21 +313,56 @@ def inspect(general_config, provider_uri, config_root, registry_infile, deployer
 @group_actor_options
 @option_target_address
 @option_ignore_deployed
+@option_confirmations
 @click.option('--retarget', '-d', help="Retarget a contract's proxy.", is_flag=True)
 @click.option('--multisig', help="Build raw transaction for upgrade via MultiSig ", is_flag=True)
-def upgrade(general_config, actor_options, retarget, target_address, ignore_deployed, multisig):
-    """
-    Upgrade NuCypher existing proxy contract deployments.
-    """
-    # Init
-    emitter = general_config.emitter
+def upgrade(general_config, actor_options, retarget, target_address, ignore_deployed, multisig, confirmations):
+    """Upgrade NuCypher existing proxy contract deployments."""
 
-    ADMINISTRATOR, _, _, registry = actor_options.create_actor(emitter, is_multisig=bool(multisig))  # FIXME: Workaround for building MultiSig TXs
+    #
+    # Setup
+    #
+
+    emitter = general_config.emitter
+    ADMINISTRATOR, deployer_address, blockchain, local_registry = actor_options.create_actor(emitter, is_multisig=bool(multisig))  # FIXME: Workaround for building MultiSig TXs | NRN
 
     contract_name = actor_options.contract_name
     if not contract_name:
         raise click.BadArgumentUsage(message="--contract-name is required when using --upgrade")
 
+    github_registry = establish_deployer_registry(emitter=emitter, download_registry=True, network=actor_options.network)
```

Comment:
This can be problematic, as it overwrites the default registry, which we don't know if can have local changes or not. Maybe we can prompt and ask for confirmation before doing this?

Reply:
I don't think it overrides the on-disk registry:

```python
    if download_registry:
        registry = InMemoryContractRegistry.from_latest_publication(network=network)
        emitter.message(PRODUCTION_REGISTRY_ADVISORY.format(source=registry.source))
        return registry
```

Reply:
I see. Thanks fine then!

Reply:
:heavy_check_mark: